play/square-wars/script.js

```diff
@@ -0,0 +1,715 @@
+//global objects for the game
+let player, enemy, enemys, bullet, bullets, boss, gameHandler, mathHandler;
+//global boolean for the game engine
+let gamePlaying;
+
+//difficulty level for game preset to 1
+let difficultyLevel = 1
+
+/**
+* used to set the difficulty level from the html main menu buttons
+* difficultyLevel is used to set up the MathClassHandler
+* @param {String} Level
+**/
+function setLevel(level) {
+  intLevel = parseInt(level)
+  difficultyLevel = intLevel;
+  document.getElementById("gameBtn").style.display = "inline";
+}
+
+/**
+* Fucntion to validate the numbers only in the input text box
+* @param (Event) evt
+**/
+function isNumber(evt) {
+  var iKeyCode = (evt.which) ? evt.which : evt.keyCode
+  if (iKeyCode != 46 && iKeyCode > 31 && (iKeyCode < 48 || iKeyCode > 57)) {
+    return false;
+  }
+  return true;
+ }
+
+
+/**
+  MoveableCanvasObject class is all objects that are moveable on the canavs
+**/
+class MoveableCanvasObject {
+  constructor(width, height, x, y, color) {
+    this.width = width;
+    this.height = height;
+    this.x = x;
+    this.y = y;
+    this.color = color;
+  }
+  update() {
```

Comment:
Add new lines between the end of a function and the start of a new one to make it feel less cluttered and more readable.